### ISSUE TITLE
Bluetooth: Remove `BT_ATT_ENFORCE_FLOW`

### ISSUE THREAD

**Issue by jori-nordic** [zephyrproject-rtos/zephyr#65910]
ATT mandates that only one transaction be open at any given time per-bearer and per direction.
That means in practice only one indication, one request etc.

The current host implementation tries to enforce this by using the HCI num complete packets event as a signal that a response has been sent and that transaction is over.
This is not compatible with all controllers: see https://github.com/zephyrproject-rtos/zephyr/issues/64967

What is illegal and ATT could instead try to detect is the peer pipelining ATT transactions, e.g. batching write requests before having received a response. Enforcing this is debatable, as it's not our job to validate that the peer's host follows spec.

What ATT should instead do, is error out when it is not able to allocate a resource due the peer not respecting the spec.

Validating that a response has been sent (and the transaction is over) can be done at the handover to L2CAP TX.

TLDR: The host doesn't need to validate peer behavior, it only needs to protect itself.

**Comment by jori-nordic** [comment]
Commit that originally added it for reference: https://github.com/zephyrproject-rtos/zephyr/commit/4a57bf6e6cd1e4b1fad87ff2611ff604800ec7b6
